cmonge/trainers/conditional_monge_trainer.py

```
@@ -349,15 +357,30 @@ def load_checkpoint(
         datamodule: ConditionalDataModule,
         ckpt_path: Path,
     ) -> None:
-        out_class = cls(
-            jobid=jobid,
-            logger_path=logger_path,
-            config=config,
-            datamodule=datamodule,
-        )
-        checkpointer = PyTreeCheckpointer()
-        out_class.state_neural_net = checkpointer.restore(
-            ckpt_path, item=out_class.state_neural_net
-        )
-        logger.info("Loaded ConditionalMongeTrainer from checkpoint")
-        return out_class
+        try:
+            out_class = cls(
+                jobid=jobid,
+                logger_path=logger_path,
+                config=config,
+                datamodule=datamodule,
+            )
+            if ckpt_path is None:
+                if len(config.checkpointing_path) > 0:
+                    ckpt_path = config.checkpointing_path
+                else:
+                    logger.error(
+                        """Provide checkpointing path either directly or
+                        through the model config"""
+                    )
+                    return
+            checkpointer = PyTreeCheckpointer()
+            out_class.state_neural_net = checkpointer.restore(
+                ckpt_path, item=out_class.state_neural_net
+            )
+            logger.info("Loaded ConditionalMongeTrainer from checkpoint")
+            return out_class
+        except:
```

dont use bare except

---

replaced with `except Exception`